lib/ansible/module_utils/k8s/raw.py

```diff
@@ -231,7 +237,9 @@ def perform_action(self, resource, definition):
             if self.append_hash and definition['kind'] in ['ConfigMap', 'Secret']:
                 name = '%s-%s' % (name, generate_hash(definition))
                 definition['metadata']['name'] = name
-            params = dict(name=name, namespace=namespace)
+            params = dict(name=name)
+            if namespace:
+                params['namespace'] = namespace
             existing = resource.get(**params)
```

Comment:
We probably shouldn't make this call when `apply` is set and state is `present`, as we'll be duplicating that `get` which could generate a lot of unnecessary API activity (especially for something like an operator which will be calling this a lot)

Reply:
Hmm, maybe scratch this one, as we will still need `existing` for computing the diffs.